flytepropeller/pkg/controller/handler.go

```diff
@@ -385,11 +385,24 @@ func (p *Propeller) streak(ctx context.Context, w *v1alpha1.FlyteWorkflow, wfClo
 			// Workflow is too large, we will mark the workflow as failing and record it. This will automatically
 			// propagate the failure in the next round.
 			mutableW := w.DeepCopy()
-			mutableW.Status.UpdatePhase(v1alpha1.WorkflowPhaseFailing, "Workflow size has breached threshold, aborting", &core.ExecutionError{
-				Kind:    core.ExecutionError_SYSTEM,
-				Code:    "WorkflowTooLarge",
-				Message: "Workflow execution state is too large for Flyte to handle.",
-			})
+			// catch potential indefinite update loop
+			if mutatedWf.GetExecutionStatus().IsTerminated() {
+				SetFinalizerIfEmpty(mutableW, FinalizerKey)
```

Comment:
This should be `ResetFinalizers(mutatedWf)` I believe. The existence of finalizers stop k8s from garbage collecting the resource, so we need to remove them to allow the CR to be cleaned up.

Reply:
thanks for catching this